test/token/token.test.ts

```diff
@@ -543,6 +543,183 @@ describe('A token', () => {
         await expect(signedToken.matchesKeyId(apiKey)).resolves.toBeTrue();
     });
 
+    it('should create a copy of a token with the given headers', () => {
+        const token = Token.of(
+            {
+                typ: 'JWT',
+                alg: 'none',
+            },
+            {
+                iss: 'croct.io',
+                aud: 'croct.io',
+                iat: 1440982923,
+            },
+            'signature',
+        );
+
+        const headers = {
+            typ: 'JWT',
+            alg: 'other',
+            appId: '00000000-0000-0000-0000-000000000000',
+        };
+
+        const newToken = token.withHeaders(headers);
+
+        expect(newToken.getHeaders()).toEqual(headers);
+        expect(newToken.getPayload()).toEqual(token.getPayload());
+        expect(newToken.getSignature()).toBe(token.getSignature());
+    });
+
+    it('should create a copy of a token with the given payload', () => {
+        const token = Token.of(
+            {
+                typ: 'JWT',
+                alg: 'none',
+            },
+            {
+                iss: 'croct.io',
+                aud: 'croct.io',
+                iat: 1440982923,
+            },
+            'signature',
+        );
+
+        const payload: TokenPayload = {
+            iss: 'test.io',
+            aud: 'test.io',
+            iat: 1440982923,
+            sub: 'c4r0l',
+        };
+
+        const newToken = token.withPayload({
+            ...payload,
+            // Ensure this prop is removed
+            jti: undefined,
```

Comment:
There is no logic for this in the method

Reply:
Nice catch! I was just testing to see if you were paying attention to the review 👀 

Did you know that in Jest, the assertion `expect({foo: true}).toEqual({foo: true, bar: undefined})` is considered true? It's a bad design decision, but they realized it too late to make a change without breaking existing code. So, they introduced a new assertion called `toStrictEqual` that behaves as we would expect. It's similar to the difference between `===` and `====` operators in JavaScript.